src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,172 @@
+        """
+        rand_value = torch.empty(size=shape, dtype=session.tensor_type).random_(
+            generator=gen
+        )
+        base = session.config.encoder_base
+        precision = session.config.encoder_precision
+        scale = base ** precision
+        x1, x2, x3 = ptr_list
+        x1_trunc = x1 >> precision if base == 2 else x1 // scale
+        x_trunc = (x2 + x3) >> precision if base == 2 else (x2 + x3) // scale
+        shares = [x1_trunc, x_trunc - rand_value, rand_value]
+        ptr_list = ReplicatedSharedTensor.distribute_shares(shares, session)
+        return ptr_list
+
+    @staticmethod
+    def truncate(x: MPCTensor, session: Session) -> MPCTensor:
+        """Perfoms the ABY3 truncation algorithm.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            MPCTensor: truncated MPCTensor.
+
+        Raises:
+            ValueError : parties involved in the computation is not equal to three.
+            ValueError : Invalid MPCTensor share pointers.
+
+        TODO :Switch to trunc2 algorithm  as it is communication efficient.
+        """
+        if session.nr_parties != 3:
+            raise ValueError("Share truncation algorithm 1 works only for 3 parites.")
+
+        # RSPointer - public ops, Tensor Pointer - Private ops
+        ptr_list = []
+        ptr_name = x.share_ptrs[0].__name__
+        if ptr_name == "ReplicatedSharedTensorPointer":
+            ptr_list.append(x.share_ptrs[0].get_shares()[0].get_copy())
+            ptr_list.extend(x.share_ptrs[1].get_copy().shares)
+        elif ptr_name == "TensorPointer":
+            ptr_list = [share.get_copy() for share in x.share_ptrs]
+        else:
+            raise ValueError("{ptr_name} not supported.")
+
+        share_ptrs = ABY3.truncation_algorithm1(ptr_list, x.shape, session)
+        result = MPCTensor(shares=share_ptrs, session=session, shape=x.shape)
+
+        return result
+
+    @staticmethod
+    def truncation_algorithm2(x: MPCTensor, session: Session) -> MPCTensor:
+        """Truncates the MPCTensor by scale factor using trunc2 algorithm.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            MPCTensor: truncated MPCTensor.
+
+        TODO : The truncation algorithm 2 is erroneous, to be optimized.
+        """
+        r, rPrime = ABY3.get_truncation_pair(x, session)
+        scale = session.config.encoder_base ** session.config.encoder_precision
+        # op = getattr(operator,"sub")
+        x_rp = x - rPrime
+        x_rp = x_rp.reconstruct(decode=False) // scale
+        zero = torch.tensor([0])
+        x_rp = MPCTensor(shares=[x_rp, zero, zero], session=x.session, shape=x.shape)
+
+        result = r + x_rp
+
+        return result
+
+    @staticmethod
+    def get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
```

Comment:
Q: Can we name this ```_get_truncation_pair``` (with leading `_`) only to show the "dev" that this is intended to be used only internally in this file?

Reply:
Sure,I would change it :+1: